docs/CONTRIBUTE_CODE.md

```diff
@@ -0,0 +1,140 @@
+# Contribute Code To Eventgen
+
+If you want to contribute code to eventgen, please read over the following guidelines before creating any pull request.
+
+
+## Pull request guidelines
+
+
+If you want to contribute to an eventgen repo, please use a GitHub pull request. This is the fastest way for us to evaluate your code and to merge it into the code base. Please don’t file an issue with snippets of code. Doing so means that we need to manually merge the changes in and update any appropriate tests. That decreases the likelihood that your code is going to get included in a timely manner. Please use pull requests.
+
+### Get started
+
+If you’d like to work on a pull request and you’ve never submitted code before, follow these steps:
+1. fork eventgen to your github workspace
+2. If you want to fix bugs or make enhancement, please make sure there is a issue in eventgen project. Refer [this guide](FILE_ISSUES.md) to create a issue.
+
+
+After that, you’re ready to start working on code.
+
+### Working on the code
+
+The process of submitting a pull request is fairly straightforward and generally follows the same pattern each time:
+1. Create a new branch
+2. Make your changes and check into local branch
+3. Rebase onto upstream
+4. Run the test
+5. Push your change
+6. Submit the pull request
+
+#### Step1: Create a new branch
+
+The first step to sending a pull request is to create a new branch in your eventgen fork. Give the branch a descriptive name that describes what it is you’re fixing. Although the branch name can be any words, we highly recommend you to use some descriptive and structured name. It will be good for you to manage the branches when there are many branches in your fork.
+```bash
+# The branch name contains 2 parts. First part works like a label, it describe what type of issue this branch is working on. Second part is the issue id.
+# For example, if the code change is a bug fix
+$ git checkout -b bug/issue123
+# For example, if the code change is to address change request
+$ git checkout -b change/issue123
+# For example, if the code change is only about refine the build process, such as making changes about CICD process
+$ git checkout -b build/issue123
+# For example, if the code change is only about refining the test cases or paying for tech debt
+$ git checkout -b chore/issue123
+```
+
+#### Step2: Make your changes and check into local branch
+
+Once you finished your change, commit them into your local branch.
+```bash
+$ git add -A
+$ git commit 
+```
+
+Our commit message format is as follows:
```

Comment:
define the commit message format here.
In the future, we can use CI tool to generate the release notes.